src/utils/collisions.ts

```diff
@@ -0,0 +1,44 @@
+import type { Collider, ColliderHandle, RigidBody, World } from '@dimforge/rapier3d-compat'
+import type { Group, Scene } from 'three'
+import type { Ref } from 'vue'
+
+interface CollisionSource {
+  collider: Collider
+  rigidBody: RigidBody | undefined
+};
+
+interface sourceTarget {
+  group: Group
+  rapierGroup: CollisionSource
+}
+
+export const getSourceFromColliderHandle = (world: World, handle: ColliderHandle) => {
+  const collider = world.getCollider(handle)
+  const rigidBodyHandle = collider?.parent()?.handle
+  const rigidBody
+  = rigidBodyHandle !== undefined
+    ? world.getRigidBody(rigidBodyHandle)
+    : undefined
+  const source: CollisionSource = {
+    collider,
+    rigidBody,
+  }
+
+  return source
+}
+
+export const get3DGroupFromSource = (source: CollisionSource, scene: Ref<Scene>) => {
+  const uuid = (source.rigidBody?.userData as { uuid?: string })?.uuid
+  const currentRigidBodyNode = scene.value.getObjectByProperty('uuid', uuid)
+
+  return currentRigidBodyNode
+}
+
+export const collisionEmisor = (
+  source: sourceTarget,
+  target: sourceTarget,
+  started: boolean,
+) => {
+  const collisionType = started ? 'enter' : 'exit'
+  source.group?.__vnode?.ctx.emit(`collision-${collisionType}`, { source, target })
```

Comment:
`__vnode` doesn't exist in the `Object3D` scope...

I propose we use the following implementation

```ts
 const collisionType = started ? 'enter' : 'exit';

 (source.group as any)?.__vnode?.ctx?.emit?.(`collision-${collisionType}`, { source, target })
```

Reply:
Done